File: Example/SnapshotTests/BPKPrimaryGradientSnapshotTests.m
/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018-2019 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

#import <FBSnapshotTestCase/FBSnapshotTestCase.h>

#import <Backpack/Gradient.h>

@interface BPKPrimaryGradientSnapshotTests : FBSnapshotTestCase

@end

@implementation BPKPrimaryGradientSnapshotTests

- (void)setUp {
    [super setUp];
    self.recordMode = NO;
}

- (void)testPrimaryGradient {
    BPKPrimaryGradientView *view = [[BPKPrimaryGradientView alloc] initWithFrame:CGRectMake(0, 0, 150, 200)];

    FBSnapshotVerifyView(view, nil);
}

- (void)testPrimaryGradientWithCustomGradient {
    BPKPrimaryGradientView *view = [[BPKPrimaryGradientView alloc] initWithFrame:CGRectMake(0, 0, 150, 200)];
    UIColor *startColor = [UIColor colorWithRed:0.00392 green:0.227 blue:0.463 alpha:1];
    UIColor *endColor = [UIColor colorWithRed:0 green:0.184 blue:0.38 alpha:1];
    CGPoint startPoint = [BPKGradient startPointForDirection:BPKGradientDirectionBottomRight];
    CGPoint endPoint = [BPKGradient endPointForDirection:BPKGradientDirectionBottomRight];

    view.gradient = [[BPKGradient alloc] initWithColors:@[startColor, endColor]
                                             startPoint:startPoint
                                               endPoint:endPoint];

    FBSnapshotVerifyView(view, nil);
}

- (void)testPrimaryGradientWithDirectionOverride {
    BPKPrimaryGradientView *view = [[BPKPrimaryGradientView alloc] initWithFrame:CGRectMake(0, 0, 150, 200)];
    view.directionOverride = @(BPKGradientDirectionLeft);

    FBSnapshotVerifyView(view, nil);
}

@end

File: Backpack/Tests/SnapshotTests/BPKPrimaryGradientSnapshotTests.swift
/*
 * Backpack - Skyscanner's Design System
 *
 * Copyright 2018-2022 Skyscanner Ltd
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import XCTest
import Backpack
import SnapshotTesting

final class BPKPrimaryGradientSnapshotTests: XCTestCase {

    override func setUpWithError() throws {
        try? super.setUpWithError()
        isRecording = false
    }
    
    func testPrimaryGradient() {
        // Given
        let sut = BPKPrimaryGradientView()
        
        // When
        sut.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            sut.widthAnchor.constraint(equalToConstant: 150),
            sut.heightAnchor.constraint(equalToConstant: 200)
        ])
        
        // Then
        assertSnapshot(sut)
    }
    
    func testPrimaryGradientWithCustomGradient() {
        // Given
        
        let startColor = BPKColor.statusSuccessFillColor
        let endColor = BPKColor.statusWarningFillColor
        
        let startPoint = BPKGradient.startPoint(for: .bottomRight)
        let endPoint = BPKGradient.endPointFor(for: .bottomRight)
        
        let sut = BPKPrimaryGradientView()
        
        // When
        sut.gradient = BPKGradient(
            colors: [
                startColor,
                endColor
            ],
            start: startPoint,
            end: endPoint
        )
        
        sut.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            sut.widthAnchor.constraint(equalToConstant: 150),
            sut.heightAnchor.constraint(equalToConstant: 200)
        ])
        
        // Then
        assertSnapshot(sut)
    }
    
    func testPrimaryGradientWithDirectionOverride() {
        // Given
        let sut = BPKPrimaryGradientView()
        sut.directionOverride = NSNumber(value: BPKGradientDirection.left.rawValue)
        
        // When
        sut.translatesAutoresizingMaskIntoConstraints = false
        NSLayoutConstraint.activate([
            sut.widthAnchor.constraint(equalToConstant: 150),
            sut.heightAnchor.constraint(equalToConstant: 200)
        ])
        
        // Then
        assertSnapshot(sut)
    }

}
